sn_node_manager/src/rpc_client.rs

```diff
@@ -0,0 +1,76 @@
+use color_eyre::eyre::bail;
+use color_eyre::{eyre::eyre, Result};
+use libp2p_identity::PeerId;
+use sn_service_management::safenode_manager_proto::safe_node_manager_client::SafeNodeManagerClient;
+use sn_service_management::safenode_manager_proto::NodeServiceRestartRequest;
+use std::net::SocketAddr;
+use std::str::FromStr;
+use std::time::Duration;
+use tonic::transport::Channel;
+use tonic::Request;
+
+struct DaemonRpcClient {
+    addr: SocketAddr,
+    rpc: SafeNodeManagerClient<Channel>,
+}
+
+pub async fn restart_node(
+    peer_id_string_vec: Vec<String>,
+    socket_addr: SocketAddr,
+    retain_peer_id: bool,
+) -> Result<()> {
+    for id_as_string in peer_id_string_vec {
+        rpc_restart(
+            id_as_string,
+            get_rpc_client(socket_addr).await?,
+            retain_peer_id,
+        )
+        .await?;
+    }
+    Ok(())
+}
+
+async fn rpc_restart(
+    id_as_string: String,
```

Comment:
Solid Principles. One function should do one thing. It helps in many ways to do this. I do it where I can, but I am very much fighting a losing battle. However I can't give up on this as I believe it's a good thing to do. I will however inline this as it's up to you.

Reply:
I don't really see how that principle would be violated by inlining this. But this one was only a suggestion. I do think it would be helpful to rename the variables though.

Reply:
It's all gone through. No worries 👍